ansible_collections/arista/avd/tests/unit/plugins/plugin_utils/test_ipaddressing.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+
+from ansible_collections.arista.avd.plugins.plugin_utils.errors import AristaAvdError
+from ansible_collections.arista.avd.plugins.plugin_utils.ip_addressing_utils import get_ip_from_pool
+
+# default values for testcases
+(pool, prefixlen, subnet_offset, ip_offset) = ("1.2.3.4/24", 32, 1, 0)
+
+
+@pytest.mark.parametrize("pool, prefixlen, subnet_offset, ip_offset, expected", [(pool, prefixlen, subnet_offset, ip_offset, "1.2.3.1")])
```

Comment:
so why do we parametrize if there is only one test case?

You can reformat this file with two functions:
one for valid test
one for invalid test with some parametrize for the four exceptions, using the expected exception string as another parameter.

To make the test have a meaningful name you can do:

```
@pytest.mark.parametrize("pool, prefixlen, subnet_offset, ip_offset, expected_raise", 
    [
        pytest.param("1.2.3.0/32", prefixlen, subnet_offset, ip_offset, <expected_raise>, id="test_ip_with_low_subnet_size")
    ]
)
def test_get_ip_from_pool_invalid(pool, prefixlen, subnet_offset, ip_offset, expected_raise):
    with pytest.raises(AristaAvdError) as exc_info:
        get_ip_from_pool(pool, prefixlen, subnet_offset, ip_offset)
    assert str(exc_info.value) == expected_raise
```

Reply:
Update the code.
Thanks